common/containers/Tabs/SendTransaction/index.tsx

```diff
@@ -465,6 +465,10 @@ export class SendTransaction extends React.Component<Props, State> {
 
   public async getFormattedTxFromState(): Promise<TransactionWithoutGas> {
     const { wallet } = this.props;
+    if (wallet.isReadOnly) {
```

Comment:
Instead of throwing when the user tries to generate a tx, instead the fields can just all be disabled. 
`transaction-refactor` completely re-writes this component so perhaps i can help implement it there

Reply:
In a future PR, I want to not even show this form. I skipped over that now for exactly that `transaction-refactor` branch.